include/oneapi/dpl/internal/dynamic_selection_impl/sycl_backend.h

```diff
@@ -184,60 +184,44 @@ class sycl_backend
         constexpr bool report_task_time = report_value_v<SelectionHandle, execution_info::task_time_t, report_duration>;
 
         auto q = unwrap(s);
+
         if constexpr (report_task_submission)
-        {
             report(s, execution_info::task_submission);
-        }
+
         if constexpr (report_task_completion || report_task_time)
         {
-            report_clock_type::time_point t0;
-            if constexpr (report_task_time)
-            {
-                if (!is_profiling_enabled)
-                {
-                    t0 = report_clock_type::now();
-                }
-            }
+            const auto t0 = report_clock_type::now();
```

Comment:
We only need to calculate t0 when report_task_time is true and is_profile_enabled is called

Reply:
I believe we are able to calculate it every once, because it's very lightweight operation.
But use it when it's really required.